### PR TITLE
trouble flag on period tabs

### PR DESCRIPTION
**Pull request by darrenmason**
This is regarding task: https://www.pivotaltracker.com/story/show/102331900

image

### CODE REVIEW

**Comment by philschatz** [comment]
@Fredasaurus should the trouble flag be on the section as well as the period?

**Comment by Fredasaurus** [comment]
Well, to me the class period with the issue should be obvious when looking at the progress indicators, so it wouldn't be needed -- YMMV.

**Comment by Dantemss** [comment]
I think Phil meant the Page :)

File: resources/styles/mixins.less [hunk 1]
```diff
@@ -90,6 +90,18 @@
         color: @tutor-neutral-darker !important;
       }
     }
+    
+    &.is-trouble {
+      a {
+      display: inline-block;
```
Comment (philschatz) [inline resources/styles/mixins.less]: *minor:* these rules should be indented. Should there be a class on `<a>`?

File: src/components/course-periods-nav.cjsx [hunk 2]
```diff
@@ -57,7 +58,14 @@ CoursePeriodsNav = React.createClass
     @setState(active: key)
 
   renderPeriod: (period, key) ->
-    <BS.NavItem eventKey={key} key="period-nav-#{period.id}">{period.name}</BS.NavItem>
+    {plan} = @props
+    isTrouble = _.pluck(plan.stats, 'is_trouble')
```
Comment (philschatz) [inline src/components/course-periods-nav.cjsx]: It looks like this puts a trouble flag on all periods if any contain `is_trouble`. I do not think this is the desired behavior.

File: src/components/course-periods-nav.cjsx [hunk 3]
```diff
@@ -57,7 +58,14 @@ CoursePeriodsNav = React.createClass
...
+    <BS.NavItem 
+      className={'is-trouble' if _.contains(isTrouble, true)}
```
Comment (philschatz) [inline src/components/course-periods-nav.cjsx]: should this be `className={'is-trouble' if period.is_trouble}`?

If so, then is the `plan` prop used anymore?

**Comment by darrenmason** [comment]
switched back to period.is_trouble.  Testing with https://github.com/openstax/tutor-server/pull/679

**Pull request merged**